### PR TITLE
Feature: Display banner when recent items are turned off from File Explorer

### LINKED ISSUE #5882: Recent items shouldn't register elements when you have it disabled
### What's the Problem?

If you have disabled recent items, when you enable it back it should show an empty list

### Solution/Idea

When you disable recent items, it should automatically clear the list and not register new recent items until you enable it back.

### Alternatives

Adding a simple setting about disabling the items history, so if you have hidden recent items but you have disabled that setting, it will still work as it does now.

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish clear the list of recent items when disabling it | Must |
| This proposal will allow developers to accomplish not register new recent items when it's disabled | Must |

### Files Version

1.9.25.0 

### Windows Version

Windows 10 21H1 10.0.19043.1165

### Additional comment

_No response_

### PR DESCRIPTION
**Pull request by jiejasonliu**
**Resolved / Related Issues**
- Closes #5882 

**Details of Changes**
- show warning banner when setting is disabled in windows
- slight refactor to empty recent files text 

**Validation**
- [X] Built and ran the app
- [X] Tested the changes for accessibility

**Screenshots (optional)**

https://user-images.githubusercontent.com/29434693/208609752-ef132e6a-02a9-4633-b880-8f23329a82ab.mp4

### CODE REVIEW

**Review by QuaintMako — DISMISSED** [review]
LGTM.

**Comment by yaira2** [comment]
Should the other message be hidden in this scenario?

**Comment by jiejasonliu** [comment]
> Should the other message be hidden in this scenario?

i initially had it that way, but wasn't sure. it's been changed now.

Files-5882-hide-empty

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml [hunk 1]
```diff
@@ -15,14 +15,22 @@
...
+			Visibility="{x:Bind IsRecentFilesDisabledInWindows, Mode=OneWay}" />
 		<TextBlock
 			x:Name="RecentItemDescription"
 			HorizontalAlignment="Stretch"
 			FontSize="14"
 			Text="{helpers:ResourceString Name=RecentItemDescription/Text}"
 			TextAlignment="Center"
 			TextWrapping="WrapWholeWords"
-			Visibility="{x:Bind Empty.Visibility, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />
+			Visibility="{x:Bind EmptyRecentsTextVisibility, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />
```
Comment (yaira2) [inline src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:33]:
```suggestion
			Visibility="{x:Bind EmptyRecentsTextVisibility, Mode=OneWay, UpdateSourceTrigger=PropertyChanged}" />
```

Comment (jiejasonliu) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml]: good catch, no clue why it was ever `TwoWay`

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs [hunk 2]
```diff
@@ -53,11 +81,22 @@ public RecentFilesWidget()
 			refreshRecentsCTS = new CancellationTokenSource();
 
 			// recent files could have changed while widget wasn't loaded
-			_ = App.RecentItemsManager.UpdateRecentFilesAsync();
+			 _ = RefreshWidget();
```
Comment (yaira2) [inline src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs:84]: Extra space?

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml (hunk 1 above)
Comment (yaira2) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml]: Seems to be because of UpdateSourceTrigger

**Comment by yaira2** [comment]
image
@jiejasonliu how did you get the icon in the tab to show?

**Comment by yaira2** [comment]
> show warning banner when setting is disabled in windows

Is the banner supposed to show when there aren't any recent items?

**Comment by yaira2** [comment]
The banner seems to always be open

**Comment by jiejasonliu** [comment]
> The banner seems to always be open

that was the intention (if you disabled the setting), but what should be the better behavior?
we could make the banner closable and restore the text, but that raises questions on if it should be displayed once only

**Comment by yaira2** [comment]
It's showing for me even when the setting is enabled

**Comment by jiejasonliu** [comment]
> It's showing for me even when the setting is enabled

one thing note is you'll have to manually refresh the widget (or navigate away) for the change to update
that said, I cannot seem to reproduce this, but it'll likely be related to the registry values

@QuaintMako if you have a chance, could you try reproducing this on your machine?
you can modify File Explorer's recent files settings by clicking the `ellipsis menu > Options` as seen here:
image

**Comment by QuaintMako** [comment]
> @QuaintMako if you have a chance, could you try reproducing this on your machine? you can modify File Explorer's recent files settings by clicking the `ellipsis menu > Options` as seen here: image

Sure thing, will try to do it soon. If I don't report in two or three days, ping me again please!

**Comment by jiejasonliu** [comment]
@yair100 i believe the issue is resolved; it was due to an incorrect default value (registry keys don't appear unless changed once)

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml [hunk 3]
```diff
@@ -15,14 +15,22 @@
 		MinHeight="200"
 		Orientation="Vertical"
 		Spacing="4">
+		<muxc:InfoBar
+			Margin="0,0,0,12"
```
Comment (yaira2) [inline src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:19]: What's the margin for?

Comment (jiejasonliu) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:19]: it was previously for when the banner and the empty text were shown together, but maybe we should keep it

the info banner is pull based; so there are some cases where they _temporarily_ show together (until a refresh occurs)

Comment (yaira2) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:19]: I recommend removing it for simplicity but if you think it's better, we can leave as is.

File: src/Files.App/Filesystem/RecentItems.cs [hunk 4]
```diff
@@ -204,6 +205,46 @@ public Task<bool> UnpinFromRecentFiles(RecentItem item)
...
+				bool startTrackDocsValue = Convert.ToBoolean(advSubkey.GetValue("Start_TrackDocs", true)); // 1 by default
+				if (!startTrackDocsValue)
+				{
+					return false;
+				}
+			}
+
+			// for users in group policies
+			using var userPolicySubkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer");
+			using var sysPolicySubkey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer");
```
Comment (yaira2) [inline src/Files.App/Filesystem/RecentItems.cs:234]: Might be good to move to the top of the method. Same for the other usings.

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml (hunk 3 above)
Comment (jiejasonliu) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml]: i decided to change the logic to make them truly mutually exclusive so they cannot ever show together

there's no need for the margin anymore!

File: src/Files.App/Filesystem/RecentItems.cs (hunk 4 above)
Comment (jiejasonliu) [reply src/Files.App/Filesystem/RecentItems.cs]: done! considering most people have recent items on, the happy path goes through all conditionals anyway

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**